### PR TITLE
Release gem GH action

### PR DESCRIPTION
**Pull request by stejskalleos**

### CODE REVIEW

**Comment by nadjaheitmann** [comment]
I like the topic that much that I opened another PR for this :see_no_evil:  https://github.com/theforeman/foreman_puppet/pull/395 

I think, we should merge one of them :)

File: .github/workflows/release_gem.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: Release
+
+on:
+  create:
+    ref_type: tag
```
Comment (evgeni) [inline .github/workflows/release_gem.yml:5]:
```suggestion
  push:
    tags:
      - 'v*'
```

It's been said this is better (see https://github.com/theforeman/actions/?tab=readme-ov-file#gem-release)

File: .github/workflows/release_gem.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+jobs:
+  release:
+    name: Release gem
+    uses: theforeman/actions/.github/workflows/release-gem.yml@v0
+    with:
+      allowed_owner: theforeman
+    secrets:
+      api_key: ${{ secrets.RUBYGEM_API_KEY }}
```
Comment (evgeni) [inline .github/workflows/release_gem.yml:14]: created

**Review by evgeni — CHANGES_REQUESTED** [review]

File: .github/workflows/release_gem.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+name: Release
```
Comment (nadjaheitmann) [inline .github/workflows/release_gem.yml:1]: I wonder why the github actions are never prepended with `---`  ?

Comment (evgeni) [reply .github/workflows/release_gem.yml:1]: people being lazy? :)

Comment (nadjaheitmann) [reply .github/workflows/release_gem.yml:1]: I propose to add another action as done here, then... ;) https://github.com/theforeman/smart_proxy_salt/blob/master/.github/workflows/yamllint.yaml

Comment (stejskalleos) [reply .github/workflows/release_gem.yml:2]: > people being lazy? :)

Yes.

**Review by evgeni — APPROVED** [review]

**Pull request merged**